browser/ui/split_view/split_view_controller.h

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_SPLIT_VIEW_SPLIT_VIEW_CONTROLLER_H_
+#define BRAVE_BROWSER_UI_SPLIT_VIEW_SPLIT_VIEW_CONTROLLER_H_
+
+#include <optional>
+#include <vector>
+
+#include "base/memory/raw_ptr.h"
+#include "base/memory/raw_ref.h"
+#include "base/scoped_observation.h"
+#include "brave/browser/ui/split_view/split_view_web_panel_data.h"
+#include "brave/browser/ui/tabs/split_view_browser_data.h"
+#include "brave/browser/ui/tabs/split_view_browser_data_observer.h"
+#include "components/tab_collections/public/tab_interface.h"
+
+class SplitViewBrowserData;
+class SplitViewView;
+class SplitViewWebPanelData;
+class TabStripModel;
+struct TabTile;
+
+namespace content {
+class WebContents;
+}  // namespace content
+
+// Control split view state based on two models(split view tab tile and web
```

Comment:
This description is not really helpful because I have no idea what "control state" means. What specific state does it manage?